src/messageParser.ts

```diff
@@ -76,6 +59,25 @@ export class Parser {
 					placeholderLevel = nestingLevel;
 				}
 			},
+			onLiteralValue: (value: any, offset: number) => {
+				if (inReferenceTag) {
+					isReference = (value === true);
+					inReferenceTag = false;
+				} else if (nestingLevel === 1 && messageKey !== null) {
+					try {
+						var message = parseMessage(value, offset, false);
+						messages.set(messageKey!, message);
+					} catch (error: any) {
+						//Very hacky solution to catch all errors here and store them, but better than not checking at all... The error has no special type, unfortunately.
+						if (String(error).startsWith('Error: Unbalanced ')) {
+							errors.push(new Literal(String(error), offset + 1, offset + value.length + 1));
+						} else {
+							throw error;
+						}
+					}
+					messageKey.endOfMessage = offset + value.length + 2;
```

Comment:
Is it certain that +2 will be reliable? For ex. the last item won't have a trailing comma, and the user could have different whitespace/newlines/indenting than is common.

Reply:
The `+2` is caused by the start and end quotation mark in JSON, I believe.

Reply:
Oh, I see - I had mistead this as being "end + 2" and thought you were accounting for quotes and comma. I see the comma is at the start of the edit you're producing. Sorry for the confusion :D